crates/hir-def/src/body.rs

```diff
@@ -53,6 +69,20 @@ pub struct Body {
     pub body_expr: ExprId,
     /// Block expressions in this body that may contain inner items.
     block_scopes: Vec<BlockId>,
+
+    /// A map from binding to its hygiene ID.
+    ///
+    /// Bindings that don't come from macro expansion are not allocated to save space, so not all bindings appear here.
+    /// If a binding does not appear here it has `SyntaxContextId::ROOT`.
+    ///
+    /// Note that this may not be the direct `SyntaxContextId` of the binding's expansion, because transparent
+    /// expansions are attributed to their parent expansion (recursively).
+    binding_hygiene: FxHashMap<BindingId, HygieneId>,
+    /// A map from an variable usages to their hygiene ID.
+    ///
+    /// Expressions that can be recorded here are single segment path, although not all single segments path refer
+    /// to variables and have hygiene (some refer to items, we don't know at this stage).
+    expr_hygiene: FxHashMap<ExprId, HygieneId>,
```

Comment:
I guess this wouldnt be necessary if our `Name`s were storing the `SyntaxContextId`? Given path expressions would then have hygienic paths, bindings also already containing `Name`s, is there a reason we are doing it this way instead of giving `Name`'s its `SyntaxContextId` (which currently is just a `()`)?

Reply:
Yes, this won't be necessary, but this is the entire point of this PR - giving every `Name` hygiene, while required for def-site hygiene, is likely to be much more intrusive and introduce performance penalty, so we instead do it only for variables/labels.

Reply:
Ah okay, that was the reasoning behind it gotcha

Reply:
Unresolving this once, can we drop a FIXME *somewhere* talking about this decision just so if someone comes in and fully implements hygiene that they know why this special stuff exists right now?

Reply:
Done.